File: CS/DxSample/MainForm.cs
using System;
using System.Configuration;
using System.Data;
using System.Data.SqlClient;
using DevExpress.XtraBars;
using DevExpress.XtraEditors;

namespace DxSample {
    public partial class MainForm : XtraForm {
        private SqlDataAdapter DataAdapter;

        public MainForm() {
            InitializeComponent();
            OrdersGridView.OptionsCustomization.AllowFilter = false;
            OrdersGridView.OptionsCustomization.AllowGroup = false;
            OrdersGridView.OptionsCustomization.AllowSort = false;
            OrdersGridView.OptionsFind.AllowFindPanel = false;
            OrdersGridView.OptionsView.ShowGroupPanel = false;
        }

        private void MainForm_Load(object sender, EventArgs e) {
            string connectionString = ConfigurationManager.ConnectionStrings["ConnectionString"].ConnectionString;
            string selectCommandText = @"
select OrderID, OrderDate, Freight, ShipName
from Orders
order by OrderID
offset @offset rows
fetch next @size rows only
";
            this.DataAdapter = new SqlDataAdapter(selectCommandText, connectionString);
            this.DataAdapter.SelectCommand.Parameters.Add("@offset", SqlDbType.Int);
            this.DataAdapter.SelectCommand.Parameters.Add("@size", SqlDbType.Int);
            this.DataAdapter.SelectCommand.Parameters["@offset"].Value = 0;
            this.DataAdapter.SelectCommand.Parameters["@size"].Value = PageSize;

            this.DataAdapter.Fill(this.dtOrders);

        }

        private int fPageSize = 10;
        public int PageSize {
            get { return fPageSize; }
            set { fPageSize = value; }
        }

        private void EmbeddedNavigator_ButtonClick(object sender, NavigatorButtonClickEventArgs e) {
            if (e.Button.ButtonType != NavigatorButtonType.Custom) return;
            e.Handled = true;
            int offset = (int)DataAdapter.SelectCommand.Parameters["@offset"].Value;
            string tag = (string)e.Button.Tag;
            switch(tag) {
                case "NextPage":
                    offset += PageSize;
                    break;
                case "PrevPage":
                    offset -= PageSize;
                    break;
            }
            offset = Math.Max(0, offset);
            OrdersGrid.EmbeddedNavigator.Buttons.CustomButtons[0].Enabled = offset > 0;
            DataAdapter.SelectCommand.Parameters["@offset"].Value = offset;
            dtOrders.Clear();
            DataAdapter.Fill(dtOrders);
            OrdersGrid.EmbeddedNavigator.Buttons.CustomButtons[1].Enabled = dtOrders.Rows.Count == PageSize;
        }

        private void PageSizeEditor_EditValueChanged(object sender, EventArgs e) {
            BarEditItem editor = (BarEditItem)sender;
            PageSize = int.Parse((string)editor.EditValue);
            DataAdapter.SelectCommand.Parameters["@size"].Value = PageSize;
            dtOrders.Clear();
            DataAdapter.Fill(dtOrders);
        }
    }
}

File: VB/DxSample/MainForm.vb
Imports System
Imports System.Configuration
Imports System.Data
Imports System.Data.SqlClient
Imports DevExpress.XtraBars
Imports DevExpress.XtraEditors

Namespace DxSample
	Partial Public Class MainForm
		Inherits XtraForm

		Private DataAdapter As SqlDataAdapter

		Public Sub New()
			InitializeComponent()
			OrdersGridView.OptionsCustomization.AllowFilter = False
			OrdersGridView.OptionsCustomization.AllowGroup = False
			OrdersGridView.OptionsCustomization.AllowSort = False
			OrdersGridView.OptionsFind.AllowFindPanel = False
			OrdersGridView.OptionsView.ShowGroupPanel = False
		End Sub

		Private Sub MainForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			Dim connectionString As String = ConfigurationManager.ConnectionStrings("ConnectionString").ConnectionString
			Dim selectCommandText As String = "
select OrderID, OrderDate, Freight, ShipName
from Orders
order by OrderID
offset @offset rows
fetch next @size rows only
"
			Me.DataAdapter = New SqlDataAdapter(selectCommandText, connectionString)
			Me.DataAdapter.SelectCommand.Parameters.Add("@offset", SqlDbType.Int)
			Me.DataAdapter.SelectCommand.Parameters.Add("@size", SqlDbType.Int)
			Me.DataAdapter.SelectCommand.Parameters("@offset").Value = 0
			Me.DataAdapter.SelectCommand.Parameters("@size").Value = PageSize

			Me.DataAdapter.Fill(Me.dtOrders)

		End Sub

		Private fPageSize As Integer = 10
		Public Property PageSize() As Integer
			Get
				Return fPageSize
			End Get
			Set(ByVal value As Integer)
				fPageSize = value
			End Set
		End Property

		Private Sub EmbeddedNavigator_ButtonClick(ByVal sender As Object, ByVal e As NavigatorButtonClickEventArgs) Handles OrdersGrid.EmbeddedNavigator.ButtonClick
			If e.Button.ButtonType <> NavigatorButtonType.Custom Then
				Return
			End If
			e.Handled = True
			Dim offset As Integer = DirectCast(DataAdapter.SelectCommand.Parameters("@offset").Value, Integer)
			Dim tag As String = CStr(e.Button.Tag)
			Select Case tag
				Case "NextPage"
					offset += PageSize
				Case "PrevPage"
					offset -= PageSize
			End Select
			offset = Math.Max(0, offset)
			OrdersGrid.EmbeddedNavigator.Buttons.CustomButtons(0).Enabled = offset > 0
			DataAdapter.SelectCommand.Parameters("@offset").Value = offset
			dtOrders.Clear()
			DataAdapter.Fill(dtOrders)
			OrdersGrid.EmbeddedNavigator.Buttons.CustomButtons(1).Enabled = dtOrders.Rows.Count = PageSize
		End Sub

		Private Sub PageSizeEditor_EditValueChanged(ByVal sender As Object, ByVal e As EventArgs) Handles PageSizeEditor.EditValueChanged
			Dim editor As BarEditItem = DirectCast(sender, BarEditItem)
			PageSize = Integer.Parse(CStr(editor.EditValue))
			DataAdapter.SelectCommand.Parameters("@size").Value = PageSize
			dtOrders.Clear()
			DataAdapter.Fill(dtOrders)
		End Sub
	End Class
End Namespace
